### PR TITLE
Initial work on a Razor syntax

### PR DESCRIPTION
**Pull request by keith-hall**
Hi,

I did a little work on support for Razor syntax `.cshtml` files, and thought I'd share it here. It's not perfect, probably one would need to use [YAMLMacros](https://packagecontrol.io/packages/YAMLMacros) to extend the `HTML` syntax to get it to handle C# code in HTML attributes properly etc., but I believe it is better than not having Razor syntax highlighted at all.

### CODE REVIEW

**Comment by michaelblyons** [comment]
Thanks, I'll check it out. I can also push the two Razors I have if you want to look at them. One is original and the other a port with some tweaks, but neither of them is very good.

**Comment by keith-hall** [comment]
> I can also push the two Razors I have if you want to look at them.

could be useful, yes please - even if you think they are not very good, there may be something valuable there :)

**Comment by michaelblyons** [comment]
Okay, both branches are pushed. Be forewarned: they're embarrassingly bad and none of the code was written since the `embed` keyword existed. 😉 It also turns out that I didn't get very far on the port of https://packagecontrol.io/packages/Razor

File: razor.sublime-syntax [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        - include: scope:source.cs#line_of_code_in
+    #- match: '@(?=\w+(?:\.\w+)*)(?!\s*[+=|(])' # TODO: support a.b[2] etc.
+    #  comment: property/variable access
+    #  scope: punctuation.section.embedded.line.cshtml
+    #  #push: razor_block # TODO: stop at closing " etc.
+    - match: '@'
+      scope: punctuation.section.embedded.line.cshtml
+      push: razor_block
+
+  razor_block_prototype:
```
Comment (michaelblyons) [inline razor.sublime-syntax]: Do you mind using "kabob-case" for context names? Also lines 120, 124, 134.

File: razor.sublime-syntax [hunk 2]
```diff
@@ -0,0 +1,136 @@
+%YAML 1.2
+---
+# http://www.sublimetext.com/docs/3/syntax.html
+name: HTML Razor (C#)
+file_extensions:
+  - cshtml
```
Comment (michaelblyons) [inline razor.sublime-syntax]: Do you think we should take `cshtml` out of the original `cshtml.sublime-syntax` file's extensions? Or leave it for now?

File: tests/syntax_test_cshtml.cshtml [hunk 3]
```diff
@@ -0,0 +1,210 @@
+<!-- SYNTAX TEST "Packages/HTML (C#)/razor.sublime-syntax" -->
+<!-- https://docs.microsoft.com/en-us/aspnet/core/mvc/views/razor?view=aspnetcore-2.1 -->
+@using Michael.Blyons.Example
```
Comment (michaelblyons) [inline tests/syntax_test_cshtml.cshtml]: Cute, but let's do a generic `System.Collections.Generic` or something. 😉

**Review by michaelblyons — CHANGES_REQUESTED** [review]
This is certainly better and more thorough than what I had going on. I noted a couple of minor tweaks, after which I will happily merge.

More generally, and somewhat independent from this PR, do you have thoughts about if/where there should be `source` scopes? I highlight `text source` with a slightly different background to make sure I don't miss embedded stuff. It was conspicuously absent here, but in a block like

```
@if (value % 2 == 0)
{
    <p>The value was even: @value</p>
}
```

I'm not completely convinced I know which bits should have it.

File: razor.sublime-syntax (hunk 2 above)
Comment (keith-hall) [reply razor.sublime-syntax]: I wasn't sure it this extension is used more commonly for Razor or not, so I opted to leave it for this PR so as not to break anything for anyone, but for my use case I would like it removed from the original file

File: razor.sublime-syntax (hunk 1 above)
Comment (keith-hall) [reply razor.sublime-syntax]: done :)

File: tests/syntax_test_cshtml.cshtml (hunk 3 above)
Comment (keith-hall) [reply tests/syntax_test_cshtml.cshtml]: that certainly is a more real world namespace; somehow I had no imagination at the time, sorry! changed :)

**Comment by keith-hall** [comment]
Certainly the C# code should get the `source` scope, but probably not the embedded html in those blocks. Not sure how I missed that, I will take a look when I get chance. Should we also include the `@` characters, do you think? it may look odd if we don't?

File: razor.sublime-syntax (hunk 2 above)
Comment (michaelblyons) [reply razor.sublime-syntax]: Long-term, I intend to peel `cshtml` out (along with basically all the file extensions in the list) into their own files. Razor seems like the generally accepted best choice for `cshtml`.

**Comment by michaelblyons** [comment]
Sorry for ignoring this so long. Thank you, Keith!

**Pull request merged**